### PR TITLE
feat(aws-cdk): add support for user provided repository name to DockerImageAsset

### PR DESCRIPTION
**Pull request by alex-berger**
Add support for (optional) user provided repository name to DockerImageAsset. This change makes it easier to use (reference) Docker Images created and deployed by CDK from EKS (Kubernetes) YAML resource files. In contrast to ECS/Fargate Tasks one cannot directly pass dynamically created ECR repository names to Kubernetes Pod resource definitions (YAML files) as these resources are not managed by CloudFormation. This PR allows users to use predefined (well-known) ECR repository names, which can then be refered to from within Kubernetes Pod resource definitions using the `latest` image tag. For example the Docker Image created by:

```
new DockerImageAsset(this, 'MyImage, {
   directory: path.join(__dirname + "images"  , 'my-image'),
   repositoryName: "my/ecr/repository"
});
```

Can then be referenced from a Kubernetes Pod resource definition like this:

```YAML
spec:
  containers:
  - name: MyContainer
    image: 000000000000.dkr.ecr.eu-central-1.amazonaws.com/my/ecr/repository:latest
```

Note, this changes do not affect the synthesized CloudFormation template as the repository is passed 
into the stack as parameter. Futhermore, there are not unit tests as I was not able to figure out how to
add unit test for `ToolkitInfo`. I tested this new feature manually and it worked well, pusing the resutling Docker Image into an ECR repository with the expected repository name.

----

### Pull Request Checklist

* [ ] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [ ] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
* [ ] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [ ] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/assets-docker/lib/image-asset.ts [hunk 1]
```diff
@@ -10,6 +10,15 @@ export interface DockerImageAssetProps {
...
   directory: string;
+
+  /**
+   * ECR repository name, if omitted a dynamic name is used instead.
+   * Specify this property if you need to statically address the
+   * image, e.g. from a Kubernetes Pod.
+   * Note, this is only the repository name, without the registry and
+   * the tag parts.
+   */
+  repositoryName?: string;
```
Comment (rix0rrr) [inline packages/@aws-cdk/assets-docker/lib/image-asset.ts]: Please add `@default` tag.

Comment (alex-berger) [reply packages/@aws-cdk/assets-docker/lib/image-asset.ts]: What is the default value of an optional property, `null`, `unedfined` or just no value?

Comment (rix0rrr) [reply packages/@aws-cdk/assets-docker/lib/image-asset.ts]: In `@default` we don't write the literal value that would be assigned to it by the typesystem; instead, we write the behavior that the user would see if they didn't supply the value.

**Comment by rix0rrr** [comment]
Alex, I cannot push updates to this branch, probably because it's based off of your `master` branch. Is there a checkbox to the right on the PR to give me push permissions, as seen here?
If not, can you please make a branch and update this PR to track your branch instead?

**Comment by alex-berger** [comment]
Yes `Allow edits from maintainers` is enabled on this PR, so according the Github documentation you should be able to push to my master branch.

**Pull request merged**